iowrappers/redis_client.go

```diff
@@ -112,6 +112,10 @@ func (r *RedisClient) setPlace(context context.Context, place POI.Place, wg *syn
 
 func (r *RedisClient) GetMapsLastSearchTime(context context.Context, location POI.Location, category POI.PlaceCategory, priceLevel POI.PriceLevel) (lastSearchTime time.Time, err error) {
 	redisField := strings.ToLower(strings.Join([]string{location.Country, location.AdminAreaLevelOne, location.City, string(category), strconv.Itoa(int(priceLevel))}, ":"))
+	// for places in Visit category Google Maps do not provide pricing info, this is subject to change in the future
+	if category == POI.PlaceCategoryVisit {
+		redisField = strings.ToLower(strings.Join([]string{location.Country, location.AdminAreaLevelOne, location.City, string(category)}, ":"))
+	}
 	lst, cacheErr := r.client.HGet(context, MapsLastSearchTimeRedisKey, redisField).Result()
```

Comment:
I feel the code at both places are repeated, why not add a function and call that instead ?

Reply:
may be later :)